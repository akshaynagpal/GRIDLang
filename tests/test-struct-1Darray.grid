Player good
{ 
int x; 
int y; 
string s; 
} 

int gameloop()
{ 

Player good p1;
Player good[2] parray; 
p1.x = 3;
p1.y = 4;
p1.s = "1D array of structs work";
parray[0] = p1;
print(parray[0].x);
print(parray[0].y);
print(parray[0].s);

return 0; 
}

int checkGameEnd()
{
  return 1;
}
