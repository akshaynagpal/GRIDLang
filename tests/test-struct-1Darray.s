	.file	"tests/test-struct-1Darray.ll"

	.section	".note.GNU-stack","",@progbits
