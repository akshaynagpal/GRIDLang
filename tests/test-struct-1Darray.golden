3
4
1D array of structs work